src/api/unicorns.service.ts

```diff
@@ -0,0 +1,32 @@
+import { IDatabase } from "../db";
+import { Unicorn, ResolvedUnicorn } from "./interfaces";
+import { MealsService } from "./meals.service";
+
+export class UnicornsService {
+  constructor(private db: IDatabase<"unicorns", Unicorn>, private mealsService: MealsService) {}
+
+  async getUnicornById(id: string): Promise<ResolvedUnicorn | null> {
+    const unicorn = await this.db.findOneById("unicorns", id);
+
+    if (unicorn !== null) {
+      return {
+        ...unicorn,
+        meals: await this.mealsService.getMealsByUnicornId(id),
```

Comment:
run it in parallel with the getUnicorn db query

Reply:
both ways have ups and downs, and I'm okay with it now